engine/uState.ml

```diff
@@ -260,8 +262,12 @@ let process_universe_constraints ctx cstrs =
                       else raise (UniverseInconsistency (Le, l, r, None))
                     in
                     LSet.fold fold levels local
-                else
-                  enforce_leq l r local
+              else
+                match Univ.Universe.level l with
+                | Some l ->
+                  Univ.Constraint.add (l, Le, r') local
```

Comment:
Why do we not `if check_leq then local else add` here?

Reply:
Unfolding of the code led to this AFAIR.